fastestimator/util/google_download_util.py

```
@@ -12,63 +12,126 @@
-            return value
+    if is_valid_file(destination):
+        print(f"File {destination} already exists, skipping download.")
+        return
+
+    tmp_path = destination + ".tmp"
+    ca_bundle = _get_ca_bundle()
+    for attempt in range(max_retries):
+        if is_valid_file(destination):
+            return
+        if attempt > 0:
+            wait = (2**(attempt - 1)) * random.uniform(5, 15)
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)